x-pack/plugins/cloud_security_posture/server/routes/benchmark_rules/bulk_action/utils.ts

```diff
@@ -105,17 +102,22 @@ export const muteDetectionRules = async (
   return disabledDetectionRules;
 };
 
-export const updateRulesStates = async (
+export const updateBenchmarkRulesStates = async (
   encryptedSoClient: SavedObjectsClientContract,
   newRulesStates: CspBenchmarkRulesStates
-): Promise<SavedObjectsUpdateResponse<CspSettings>> => {
-  return await encryptedSoClient.update<CspSettings>(
+): Promise<CspBenchmarkRulesStates> => {
+  if (!Object.keys(newRulesStates).length) {
+    return {};
+  }
+
+  const response = await encryptedSoClient.update<CspSettings>(
     INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
     INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID,
     { rules: newRulesStates },
     // if there is no saved object yet, insert a new SO
     { upsert: { rules: newRulesStates } }
```

Comment:
@CohenIdo Would mind clarifying the comment? What's the relationship between SO and rule states?

Reply:
The rules states are saved in SO. If it is the first rule and the object doesn't exist, a new object will be created.